### PR TITLE
Use is_symbolic instead of testing isinstance in some place

### PR DESCRIPTION
**Pull request by soulitzer**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #110529
* #110528
* #109171
* __->__ #110372
* #110371
* #110370
* #110369
* #110044

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

### CODE REVIEW

**Review by ezyang — APPROVED** [review]

File: torch/fx/experimental/symbolic_shapes.py [hunk 1]
```diff
@@ -2704,7 +2704,7 @@ def create_symbolic_sizes_strides_storage_offset(
         # we may have an unnessary shape speciliazation for y.
         def maybe_specialize_sym_int_with_hint(maybe_sym) -> int:
             assert isinstance(maybe_sym, (int, torch.SymInt))
```
Comment (Skylion007) [inline torch/fx/experimental/symbolic_shapes.py:2706]: Couldn't this assert be updated too? If is_symbolic has a typing_extensions.TypeGuard annotation at least.

Comment (soulitzer) [reply torch/fx/experimental/symbolic_shapes.py:2706]: Using is_symbolic would be testing something different than what we want here. This assert is saying that we are okay with either ints or SymInts whether or not those SymInts are symbolic.

**Comment by huydhn** [comment]
@pytorchbot drci

(Please ignore this comment, I'm testing Dr.CI)

**Pull request closed without merging**